src/us/kbase/workspace/kbase/InitWorkspaceServer.java

```diff
@@ -62,16 +69,23 @@
 import us.kbase.workspace.kbase.admin.DefaultAdminHandler;
 import us.kbase.workspace.kbase.admin.KBaseAuth2AdminHandler;
 import us.kbase.workspace.kbase.admin.WorkspaceAdministration;
-import us.kbase.workspace.kbase.admin.WorkspaceAdministration.UserValidator;
 import us.kbase.workspace.listener.ListenerInitializationException;
 import us.kbase.workspace.listener.WorkspaceEventListener;
 import us.kbase.workspace.listener.WorkspaceEventListenerFactory;
 
 public class InitWorkspaceServer {
 	
+	// TODO NOW_TYPES delegation integration tests for 6 admin type methods
+	// TODO NOW_TYPES delegation integration tests for 17 standard type methods
+	// TODO NOW_TYPES delegation integration tests for saving objects, incl type fetch errors
+	// TODO NOW_TYPES document type delegation setup, constraints, caching,
+	// and long term type server split plans in rst
+	// Note that if the bootstrap admin for each server is not also an admin in all other
+	// servers confusion will ensue. Does this matter for auth2 based admins? Check
+	
 	// TODO TEST unittests... are going to be a real pain.
 	// TODO JAVADOC
-	// TODO CODE try and clean this mess up a little. Nulls everywhere, long methods.
+	// TODO CODE try and clean this mess up. Nulls everywhere, long methods.
```

Comment:
So yeah, this class is a mess (and I find that classes that build the server dependencies are always pretty messy) and a huge pain to test other than the happy paths. It needs a revamp, but not right now.

I'm not really sure if testing these types of classes is ever not going to be a nightmare